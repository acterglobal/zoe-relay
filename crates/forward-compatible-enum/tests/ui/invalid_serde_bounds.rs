use forward_compatible_enum::ForwardCompatibleEnum;

// Test invalid serde bounds syntax - should fail to compile
#[derive(ForwardCompatibleEnum)]
#[forward_compatible(serde_serialize = 123)]  // Invalid: not a string
pub enum InvalidBounds1<T> {
    #[discriminant(0)]
    Value(T),
    
    Unknown { discriminant: u32, data: Vec<u8> },
}

fn main() {}