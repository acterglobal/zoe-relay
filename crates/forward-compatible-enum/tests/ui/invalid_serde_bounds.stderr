error[E0277]: the trait bound `T: serde::Serialize` is not satisfied
 --> tests/ui/invalid_serde_bounds.rs:4:10
  |
4 | #[derive(ForwardCompatibleEnum)]
  |          ^^^^^^^^^^^^^^^^^^^^^ the trait `Serialize` is not implemented for `T`
  |
  = note: for local types consider adding `#[derive(serde::Serialize)]` to your `T` type
  = note: for types from other crates check whether the crate offers a `serde` feature flag
  = note: required for `&T` to implement `Serialize`
note: required by a bound in `to_stdvec`
 --> $CARGO/postcard-$VERSION/src/ser/mod.rs
  |
  | pub fn to_stdvec<T>(value: &T) -> Result<std::vec::Vec<u8>>
  |        --------- required by a bound in this function
  | where
  |     T: Serialize + ?Sized,
  |        ^^^^^^^^^ required by this bound in `to_stdvec`
  = note: this error originates in the derive macro `ForwardCompatibleEnum` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T` with trait `Serialize`
  |
6 | pub enum InvalidBounds1<T: serde::Serialize> {
  |                          ++++++++++++++++++

error[E0277]: the trait bound `T: serde::Deserialize<'de>` is not satisfied
 --> tests/ui/invalid_serde_bounds.rs:4:10
  |
4 | #[derive(ForwardCompatibleEnum)]
  |          ^^^^^^^^^^^^^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `T`
  |
  = note: for local types consider adding `#[derive(serde::Deserialize)]` to your `T` type
  = note: for types from other crates check whether the crate offers a `serde` feature flag
note: required by a bound in `postcard::from_bytes`
 --> $CARGO/postcard-$VERSION/src/de/mod.rs
  |
  | pub fn from_bytes<'a, T>(s: &'a [u8]) -> Result<T>
  |        ---------- required by a bound in this function
  | where
  |     T: Deserialize<'a>,
  |        ^^^^^^^^^^^^^^^ required by this bound in `from_bytes`
  = note: this error originates in the derive macro `ForwardCompatibleEnum` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider restricting type parameter `T` with trait `Deserialize`
  |
6 | pub enum InvalidBounds1<T: serde::Deserialize<'_>> {
  |                          ++++++++++++++++++++++++
